components/sup/src/main.rs

```diff
@@ -403,6 +468,36 @@ fn sub_stop(m: &ArgMatches) -> Result<()> {
     Manager::save_spec_for(&cfg, spec)
 }
 
+fn wait_for_status(cfg: ManagerConfig,
+                   ident: PackageIdent,
+                   state: &ProcessState)
+                   -> Result<ServiceStatus> {
+    loop {
+        let stop_time = time::get_time() + Duration::seconds(5);
```

Comment:
It looks like this loop could potentially run indefinitely. Did you mean to declare the stop_time outside of the loop?